### ISSUE TITLE
arch: rework Publish to copy DB and ES records

### ISSUE THREAD

**Issue by ghukill** [WSULib/combine#83]
Until now Publish jobs have run through the same pipelines as other jobs (e.g. Harvest, Transform, Merge).  

But, if we limit the ability to run new validations or mapping, we can copy the input job for a Publish job wholesale:

  * do not re-map records for ES
    * instead, copy ES index wholesale for job that is published
  * do not run validations
    * consider copying validations 
    * or, perhaps the RecordGroup page should be enough information for the user that the *input* job for a Publish job was valid, and thus, so are the Published records.

**Comment by ghukill** [comment]
Digging in, realized that Publish jobs already copy the ES index wholesale via `core.spark.es.index_published_job()`, making this transition easier.  And, reveals that the index mapper selection from Publish job screens is not used.

**Comment by ghukill** [comment]
Scratch that - once the job was indexed to ES -- using an index mapper -- *that* index was copied to the ES Index `/published`.  So, will still need to pull the **input** job's `index_mapper` and use.

**Comment by ghukill** [comment]
Of course, it would appear that the `index_mapper` used is not saved with the `Job` instance.  Adding that.

**Comment by ghukill** [comment]
Lost the forest through the trees.

If our intent was to use the same `index_mapper` from an input job when performing a Publish, given a pure copy of the records in the DB, there is no need to re-index at all.  Similar to how we formerly copied the index from a Publish job (e.g. `j42`) to the Published ES index `published`, we can do the same by copying the ES index from the **input** job that was published.  

TL/DR: copy the ES index from input job for published job

OR, skip the indexing altogether?  Should a Publish job even have an index, given it's identical to the input job?  As opposed to copying the index twice, we could just either:
  1. not show indexed fields for a Publish job
  2. quietly port the mapped fields from the input job to the Publish job view

**Comment by ghukill** [comment]
For simplicity's sake, for now, Publish jobs:
  * copy records in the DB
  * copy ES index

There are no validations or ES mapping processes, so *considerably* faster than any other jobs.  It's not instant, but very fast.  

Closing for now.
